components/BackButton.js

```diff
@@ -0,0 +1,51 @@
+import Image from "next/image";
+import { useRouter } from "next/router";
+import styled from "styled-components";
+
+export default function BackButton({ variant, handleBack }) {
+  const router = useRouter();
+
+  function handleBackClick() {
+    if (variant === "detailspage") {
+      router.push("/");
+    } else if (variant === "profilepage") {
+      handleBack();
+    } else {
+      window.history.back();
```

Comment:
you can use the router instead: `router.back()` https://nextjs.org/docs/pages/api-reference/functions/use-router#routerback